Mathlib/Topology/Algebra/OpenSubgroup.lean

```diff
@@ -269,6 +269,40 @@ theorem isOpen_of_one_mem_interior [ContinuousMul G] (H: Subgroup G)
     (h_1_int : (1 : G) ∈ interior (H : Set G)) : IsOpen (H : Set G) :=
   isOpen_of_mem_nhds H <| mem_interior_iff_mem_nhds.1 h_1_int
 
+@[to_additive]
+lemma isClosed_of_isOpen [ContinuousMul G] (U : Subgroup G) (h : IsOpen (U : Set G)) :
+    IsClosed (U : Set G) :=
+  OpenSubgroup.isClosed ⟨U, h⟩
+
+@[to_additive]
+lemma discreteTopology [ContinuousMul G] (U : Subgroup G) (h : IsOpen (U : Set G)) :
+    DiscreteTopology (G ⧸ U) := by
+  refine singletons_open_iff_discrete.mp (fun g ↦ ?_)
+  induction' g using Quotient.inductionOn with g
+  show IsOpen (QuotientGroup.mk ⁻¹' {QuotientGroup.mk g})
+  convert_to IsOpen ((g * ·) '' U)
+  · ext g'
+    simp only [Set.mem_preimage, Set.mem_singleton_iff, QuotientGroup.eq, Set.image_mul_left]
+    rw [← U.inv_mem_iff]
+    simp
+  · exact Homeomorph.mulLeft g |>.isOpen_image |>.mpr h
+
+@[to_additive]
+lemma quotient_finite_of_isOpen [ContinuousMul G] [CompactSpace G] (U : Subgroup G)
+    (h : IsOpen (U : Set G)) : Finite (G ⧸ U) :=
+  have : CompactSpace (G ⧸ U) := Quotient.compactSpace
```

Comment:
Is this one not found automatically? Do you understand why?

Reply:
I think the `CompactSpace` one is not found, because `G ⧸ U` is syntactically not the same as `Quotient (QuotientGroup.leftRel U)`, so
```
import Mathlib

variable (G : Type*) [Group G] [TopologicalSpace G] [ContinuousMul G] [CompactSpace G]
  (U : Subgroup G)

-- works
#synth CompactSpace (Quotient (QuotientGroup.leftRel U))

-- but this does not
#synth CompactSpace (G ⧸ U)

-- and this works again
example : CompactSpace (G ⧸ U) := inferInstanceAs <|
  CompactSpace (Quotient (QuotientGroup.leftRel U))
```

Reply:
I think we should add
```
@[to_additive]
instance [CompactSpace G] (N : Subgroup G) : CompactSpace (G ⧸ N) :=
  Quotient.compactSpace
```
close to the definition of the topology on `G ⧸ N`.